scripts/autotools/libmongoc/CheckSasl.m4

```diff
@@ -1,20 +1,10 @@
 PHP_ARG_WITH([mongodb-sasl],
              [whether to enable SASL for Kerberos authentication],
-             [AS_HELP_STRING([--with-mongodb-sasl=@<:@auto/no/cyrus/gssapi@:>@],
+             [AS_HELP_STRING([--with-mongodb-sasl=@<:@auto/no/cyrus@:>@],
                              [MongoDB: Enable SASL for Kerberos authentication [default=auto]])],
              [auto],
              [no])
 
-AS_IF([test "$os_darwin" = "yes" -a \( "$PHP_MONGODB_SASL" = "gssapi" -o "$PHP_MONGODB_SASL" = "auto" \)],[
-  dnl PHP_FRAMEWORKS is only used for SAPI builds, so use MONGODB_SHARED_LIBADD for shared builds
-  if test "$ext_shared" = "yes"; then
-    MONGODB_SHARED_LIBADD="-framework GSS $MONGODB_SHARED_LIBADD"
-  else
-    PHP_ADD_FRAMEWORK([GSS])
-  fi
-  PHP_MONGODB_SASL="gssapi"
-])
```

Comment:
@kvwalker found another bug with the original implementation. Although this added the GSS framework on macOS, it didn't stop the subsequent `PKG_CHECK_MODULES` check for `krb5-gssapi` and still resulted in an error if that library wasn't found. Had we kept this, we would want to ensure that the `AS_IF([test "$PHP_MONGODB_SASL" = "gssapi" -o "$PHP_MONGODB_SASL" = "auto"],[` block later on never runs.